pkg/endpoint/policy.go

```diff
@@ -777,6 +801,83 @@ func (e *Endpoint) Regenerate(regenMetadata *regeneration.ExternalRegenerationMe
 	return done
 }
 
+// InitialPolicyComputedLocked marks computation of the initial Envoy policy done.
+// Endpoint lock must be held so that the channel is never closed twice.
+func (e *Endpoint) InitialPolicyComputedLocked() {
+	select {
+	case <-e.InitialEnvoyPolicyComputed:
+	default:
+		close(e.InitialEnvoyPolicyComputed)
+	}
+}
+
+// Compute initial policy for the endpoint.
+// regenContext.datapathRegenerationContext.policyResult.release() must be eventually called
+// to release resources held for the policy.
+func (e *Endpoint) ComputeInitialPolicy(regenContext *regenerationContext) error {
```

Comment:
I think it would be useful to document in this function's godoc why an initial policy computation is necessary.